### ISSUE TITLE
Crash when validating scene in Unity 2019.1.0b1

### ISSUE THREAD

**Issue by schodemeiss** [ssannandeji/Zenject-2019#601]
When I run "Validate Scene" from the menu or the shortcut keys, the editor immediately crashes in Unity 2019.1.0b1.

It doesn't appear to produce any logs for me to submit, otherwise I'd attach them here (any ideas where I might find some)?

However, my project does run absolutely fine - it appears to just be a problem with the validation.

It's also worth noting the validation works perfectly on the exactly same project in the same state on 2018.3.

**Comment by svermeulen** [comment]
Yep I can reproduce this too.  I'll take a look

**Comment by schodemeiss** [comment]
Can confirm still exists on 2019.1.0b2 as well.

**Comment by svermeulen** [comment]
This looks like a unity bug.  It crashes on the line `gameObj.transform.SetParent(parent, positionAndRotationWereSet);` (line 1775) in `DiContainer.cs`.  I've reproduced it down to the following code:

```csharp
var parent1 = new GameObject("parent1");
parent1.hideFlags = HideFlags.HideAndDontSave;

var go = new GameObject("go");
go.transform.SetParent(parent1.transform);

var parent2 = new GameObject("parent2");
go.transform.SetParent(parent2.transform);
```

If this is executed in edit mode then the crash occurs.  I've submitted this as a bug report to unity

**Comment by schodemeiss** [comment]
This appears to be fixed in 2019.2.0a8.
